### PR TITLE
Add data query

### PR DESCRIPTION
**Pull request by czhen-zilliz**
Main page:
image
Filter:
image
Main page with results:
image
Error page:
image

### CODE REVIEW

File: client/src/http/BaseModel.ts [hunk 1]
```diff
@@ -84,4 +84,10 @@ export default class BaseModel {
     const res = await http.post(path, data);
     return res.data.data;
   }
+
+  static async query(options: updateParamsType) {
```
Comment (nameczz) [inline client/src/http/BaseModel.ts:88]: may be we can rename vectorSearch. They looks like same

File: client/src/pages/query/Query.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    setTableLoading(true);
+    try {
+      const res = await CollectionHttp.queryData(collectionName, {
+        expr: expression,
+        output_fields: fields.map(i => i.name),
+      });
+      setTableLoading(false);
+      const result = res.data;
+      setQueryResult(result);
+    } catch (err) {
```
Comment (nameczz) [inline client/src/pages/query/Query.tsx:110]: use finally setloading false

File: client/src/pages/query/Query.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+          />
+        </div>
+        <div className="right">
+          <CustomButton className="btn" onClick={handleFilterReset}>
+            <ResetIcon classes={{ root: 'icon' }} />
+            {btnTrans('reset')}
+          </CustomButton>
+          <CustomButton
+            variant="contained"
+            disabled={!expression}
```
Comment (nameczz) [inline client/src/pages/query/Query.tsx:139]: do we need trim ?

File: client/src/pages/query/Query.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            onClick={() => handleQuery()}
+          >
+            {btnTrans('query')}
+          </CustomButton>
+        </div>
+      </div>
+      {tableLoading || queryResult ? (
```
Comment (nameczz) [inline client/src/pages/query/Query.tsx:146]: queryResult logic

File: client/src/components/advancedSearch/Filter.tsx [hunk 5]
```diff
@@ -256,9 +258,21 @@ const Filter = function Filter(props: FilterProps) {
...
+  // Reset all conditions(both current and former).
+  const handleHardReset = () => {
+    setInitConditions([]);
+    handleReset();
+  };
   const handleFallback = () => {
     setFilteredFlatConditions(initConditions);
   };
+  // Expose func
+  useImperativeHandle(ref, () => ({
```
Comment (nameczz) [inline client/src/components/advancedSearch/Filter.tsx:270]: add more comments

**Review by nameczz — CHANGES_REQUESTED** [review]

File: client/src/http/BaseModel.ts (hunk 1 above)
Comment (czhen-zilliz) [reply client/src/http/BaseModel.ts:82]: Has removed vectorSearch, use query instead.

File: client/src/pages/query/Query.tsx (hunk 3 above)
Comment (czhen-zilliz) [reply client/src/pages/query/Query.tsx:148]: There's no need. Currently expression value comes from `Advanced Filter`, not user input.

**Review by nameczz — APPROVED** [review]

**Review by Tumao727 — APPROVED** [review]

**Review by shanghaikid — APPROVED** [review]

**Pull request merged**